x-pack/plugins/advanced_ui_actions/public/drilldowns/drilldown_action_factory_context.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/**
+ * Context object used when creating a drilldown.
+ */
+export interface DrilldownActionFactoryContext<T> {
+  /**
+   * Context provided to the drilldown factory by the place where the UI is
+   * rendered. For example, for the "dashboard" place, this context contains
+   * the ID of the current dashboard, which could be used for filtering it out
+   * of the list.
+   */
+  placeContext: T;
+
+  /**
+   * List of triggers that user selected in the UI.
+   */
+  triggers: string[];
```

Comment:
This isn't used anywhere yet, right?  Maybe remove it for now if so?  Then we could actually get rid of this additional layer of `context.placeContext.embeddable` and just do `context.embeddable` which I think is easier to follow (and if/when we bring this back, we could maybe change this to 

```ts
interface DrilldownActionFactoryContext {
  triggers: string[];
}
...
export interface DrilldownDefinition<
  Config extends object = object,
  PlaceContext extends DrilldownActionFactoryContext = DrilldownActionFactoryContext,
  ExecutionContext extends object = object
> 
```

and avoid the multiple levels of nesting context types.

Reply:
It is not used in this PR, but I would not remove it just for the sake of removing things. To me it looks like a very fundamental thing for a drilldown to know to which triggers it is about to be attached to.

Lifting `placeContext` a level up also is not something I would like to do, as we definitely know we will need this context once we integrate drilldowns in any of these applications: (1) Graph; (2) ML; (3) APM; (4) sample plugin without embeddables. Having `placeContext` on that level allows us to extend drilldown context in the future.

Reply:
We have no idea when we will get around to expanding drilldowns to support multiple triggers.  Because "APPLY_FILTER_TRIGGER" captures so much of the existing use cases, it might not get prioritized for months, or a year.  Exposing unused APIs with the thought of "future use" means:
 - We don't actually know if this will solve our needs
 - It's untested, undocumented 
 - Other developers can still use it because it's part of the public API
 - There is a possibility we'll making breaking changes to the API when we eventually do use it in earnest.

Same with `placeContext`. It's doesn't actually work generically right now, it's not tested in a generic fashion, but the types let implementors think it would work generically.  Any drilldown implementor today that sets a custom `PlaceContext` is making a mistake.

Reply:
I've removed `PlaceContext` in this PR, so this `triggers` field does not exist anymore, too.

---

> We have no idea when we will get around to expanding drilldowns to support multiple triggers. Because "APPLY_FILTER_TRIGGER" captures so much of the existing use cases, it might not get prioritized for months, or a year.

We currently have two triggers already: (1) `VALUE_CLICK_TRIGGER`; (2) `SELECT_RANGE_TRIGGER`. `APPLY_FILTER_TRIGGER` is not used anywhere, if I remember it correctly.